### PR TITLE
Added TS support

### PR DESCRIPTION
**Pull request by alfetopito**
~Waterfalls into https://github.com/gnosis/safe-cmm-app-react/pull/1~ merged

~Once that's merged I'll point this PR to `master`~ pointing to master

---

Typescript setup.

Added a dummy component just so build succeeds.
Also added a dummy story:

screenshot_2020-08-10_13-41-49

### CODE REVIEW

File: .storybook/webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+module.exports = ({ config }) => {
+  config.module.rules.push({
+    test: /\.(ts|tsx)$/,
+    use: [
+      {
+        loader: require.resolve("awesome-typescript-loader"),
```
Comment (Velenir) [inline .storybook/webpack.config.js:6]: Why use `awesome-typescript-loader` here and `ts-loader` in `webpack.config`? `ts-loader doesn't work for storybook?

Comment (Velenir) [reply .storybook/webpack.config.js:6]: Why `require.resolve`? Webpack is supposed to correctly resolve paths by itself, doesn't work?

File: package.json [hunk 2]
```diff
@@ -38,6 +38,10 @@
     "@storybook/addon-links": "^5.3.19",
     "@storybook/addons": "^5.3.19",
     "@storybook/react": "^5.3.19",
+    "@types/jest": "^26.0.9",
```
Comment (Velenir) [inline package.json:41]: You may also need to include `@types/react, @types/node`

File: .storybook/webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
+module.exports = ({ config }) => {
```
Comment (Velenir) [inline .storybook/webpack.config.js:1]: Does storybook use this file by default? I always thought it only used its own config internally and through `webpackFinal()`
I would have expected to see
```js
// .storybook/main.js
module.exports = {
  webpackFinal: require('./webpack.config.js')
  ...
}
```
or something

File: .storybook/webpack.config.js (hunk 1 above)
Comment (alfetopito) [reply .storybook/webpack.config.js:6]: Errr, not sure, I copied that from https://github.com/gnosis/safe-react-components/blob/development/.storybook/webpack.config.js

Will adjust

File: .storybook/webpack.config.js (hunk 3 above)
Comment (alfetopito) [reply .storybook/webpack.config.js:1]: I couldn't find where this is magically loaded either. I based this TS setup off of https://github.com/gnosis/safe-react-components/tree/development/.storybook
Nowhere I see this file mentioned directly.

File: .storybook/webpack.config.js (hunk 1 above)
Comment (alfetopito) [reply .storybook/webpack.config.js]: f2cc4e1384f3d38031c830b8cfb5215c301c8bb4

File: .storybook/webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      {
+        loader: "url-loader",
+        query: {
+          name: "[name].[ext]",
+        },
+      },
+    ],
+  });
+
+  config.resolve.extensions.push(".ts", ".tsx", "woff2");
```
Comment (Velenir) [inline .storybook/webpack.config.js:38]: I wonder if you'll need to include [module resolution](https://github.com/gnosis/safe-cmm-app-react/blob/f2cc4e1384f3d38031c830b8cfb5215c301c8bb4/webpack.config.js#L54) here too.

You can try adding an absolute import in Test component and see if it breaks.

Comment (alfetopito) [reply .storybook/webpack.config.js:38]: Spot on!

I've done it on the follow up PR https://github.com/gnosis/safe-cmm-app-react/pull/8/files#diff-6da49b24efd584154cfb4ec759745192R40

**Pull request merged**